File: KoRadio/KoRadio.Services/Database/Product.cs
using System;
using System.Collections.Generic;

namespace KoRadio.Services.Database;

public partial class Product:ISoftDelete
{
    public int ProductId { get; set; }

    public string ProductName { get; set; } = null!;

    public string ProductDescription { get; set; } = null!;

    public decimal Price { get; set; }

    public int StoreId { get; set; }
    public int StockQuantity { get; set; } = 0;
    public bool IsOnSale { get; set; }
    public decimal? SalePrice { get; set; }



    public bool IsDeleted { get; set; }

    public byte[]? Image { get; set; }

    public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();

    public virtual ICollection<ProductsService> ProductsServices { get; set; } = new List<ProductsService>();

    public virtual Store Store { get; set; } = null!;
}

File: KoRadio/UI/ko_radio_mobile/lib/models/product.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:ko_radio_mobile/models/products_service.dart';
import 'package:ko_radio_mobile/models/store.dart';

part 'product.g.dart';

@JsonSerializable()
class Product {
  int productId;
  String? productName;
  String? productDescription;
  double? price;
  int? stockQuantity;
  bool? isOnSale;
  double? salePrice;
  bool? isOutOfStock;
  DateTime? saleExpires;
  bool? isDeleted;
  String? image;
  int? storeId;
  Store? store;
  List<ProductsServices>? productsServices;

  Product({
    required this.productId,
    
  });
  
 factory Product.fromJson(Map<String, dynamic> json) => _$ProductFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$ProductToJson(this);
}

